### DISCUSSION TITLE
Autocompletion for firewall-cmd

### DISCUSSION THREAD

**Discussion by vl-tech** [firewalld/firewalld#1419] [Ideas]
Hello Community,
My question is for the optimisation of the firewall-cmd CLI. I just played around with it and added 2 lines for argcomplete + some additional argcomplete commands on the linux machine and got an autocompletion working.
Do you think it would be possible to add 
`import argcomplete` and 
`argcomplete.autocomplete(parser)` + some instructions on autocompletion ?

It can make the script heavier but it really helps with all the options .
I have the steps to configure linux bashrc for autocompletion to work if needed.
Got a screenshot of the working version on my almalinux machine
![firewall-cmd-autocomplete](https://github.com/user-attachments/assets/6583006d-be21-4c6e-869b-636837eec8be)

Thank you
Cheers!

**Comment by erig0** [comment]
> Do you think it would be possible to add
import argcomplete and
argcomplete.autocomplete(parser) + some instructions on autocompletion ?

I don't think so. This adds a new dependency on `argcomplete`.


Is the existing bash/zsh completion not sufficient?

**Reply by vl-tech** [reply]
Apologies for the discussion. I submitted it then noticed the bash completion option. However it doesn't seem to work at 100%. In terms of suggesting options the argcomplete works better but I'll just edit my local scripts as I did with this one in case I need to do something more complicated. There is no need to add another dependency to the script as you said . 
Thank your your reply.
